config/config.go

```diff
@@ -0,0 +1,79 @@
+package config
+
+import (
+	"errors"
+	"log/slog"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type Config struct {
+	Namespace string
+	Pretty    bool
+	Level     slog.Level
+}
+type Option func(cfg *Config)
+
+// FromEnv is a config option that processes environment variables and populates the config with the contents of those vars
+var FromEnv Option = func(cfg *Config) {
+	humanLog, _ := strconv.ParseBool(os.Getenv("HUMAN_LOG"))
+	cfg.Pretty = humanLog
+
+	if s := os.Getenv("LOG_LEVEL"); s != "" {
+		if l, err := levelFromString(s); err == nil {
+			cfg.Level = l
+		}
+	}
+}
```

Comment:
Would be helpful to have a readme table of these I think. I'm torn about controlling libraries with env vars rather than with config but this is the existing convention so keep going I guess. Might want to think about namespacing them maybe.

Reply:
I have added detail to [the README for these options](https://github.com/ONSdigital/log.go/pull/44/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5). As you say, this is mainly to extend the convention already set. For the new `LOG_LEVEL` if we didn't have it as a common env var we'd have to add config to every app using the library to do it there instead which would increase the migration overhead as well as leading to possibility of inconsistency between apps. Happy to change the name if you think it useful. 
As for programmatic setting of these vars, I thought about that and that's why all options can be set via options sent to initialise. Eg. for these you can do the following…
```go
log.initialise("dp-logging-example",config.Pretty,config.Level(log.LevelWarn))
```